item-trade/packages/contracts/src/systems/ItemTradeSystem.sol

```diff
@@ -213,11 +217,12 @@ contract ItemTradeSystem is System {
    * @dev Collect the ERC-20 tokens collected by the SSU
    * @param smartObjectId The smart object id of the SSU
    */
-  function collectTokens(uint256 smartObjectId, uint256 amount) public onlyOwner(smartObjectId) {
+  function collectTokens(uint256 smartObjectId) public onlyOwner(smartObjectId) {
     ItemTradeERC20Data memory ssuData = ItemTradeERC20.get(smartObjectId);
     address tokenAddress = ssuData.tokenAddress;
 
-    IERC20(tokenAddress).transfer(ssuData.receiver, amount);
+    // Transfer only the tokens collected by the SSU
+    IERC20(tokenAddress).transfer(ssuData.receiver, ItemTradeTokenSum.get(smartObjectId));
   }
```

Comment:
After the collection, is the total token amount then reset to 0?

Reply:
Yes right good catch!